contracts/Gauge.sol

```diff
@@ -334,63 +507,55 @@ contract Gauge is IGauge {
         }
         //also get rewards from linked rewards
         if (_claimExtras) {
-            for (uint256 i = 0; i < extraRewards.length; i++) {
-                IVirtualBalanceRewardPool(extraRewards[i]).getReward(
-                    msg.sender
-                );
+            uint256 length = extraRewards.length;
+            for (uint256 i = 0; i < length; i++) {
+                IExtraReward(extraRewards[i]).getRewardFor(msg.sender);
             }
         }
     }
 
-    function donate(uint256 _amount) external returns (bool) {
-        require(_amount != 0);
-        IERC20(rewardToken).safeTransferFrom(
-            msg.sender,
-            address(this),
-            _amount
-        );
-        queuedRewards = queuedRewards + _amount;
+    /**
+     * @notice
+     * Transfer penalty to the veYFIRewardContract
+     * @dev Penalty are queued in this contract.
+     * @return true
+     */
+    function transferQueuedPenalty() external returns (bool) {
+        uint256 toTransfer = queuedPenalty;
+        queuedPenalty = 0;
+
+        IERC20(rewardToken).approve(veYfiRewardPool, toTransfer);
+        BaseGauge(veYfiRewardPool).queueNewRewards(toTransfer);
```

Comment:
this takes tokens from msg.sender...why would they ever voluntarily call it if they're being penalized?

Reply:
just checked, looks like VotingEscrow calls this possible, need to investigate it further.